apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientClaimsEobTaskTransformer.java

```diff
@@ -0,0 +1,365 @@
+    this.samhsaMatcher = requireNonNull(samhsaMatcher);
+    this.drugCodeDisplayLookup = requireNonNull(drugCodeDisplayLookup);
+    this.npiOrgLookup = requireNonNull(npiOrgLookup);
+  }
+
+  /**
+   * Setup the task processing parameters.
+   *
+   * @param claimTransformer the {@link ClaimTransformerInterface} to run.
+   * @param claimType the claim type to process.
+   * @param id the beneficiary or claim identifier to process.
+   * @param lastUpdated the date range that lastUpdate falls within.
+   * @param serviceDate the date range that clm_thru_dt falls within.
+   * @param excludeSamhsa if true excludes SAMHSA claims.
+   */
+  public void setupTaskParams(
+      ClaimTransformerInterface claimTransformer,
+      ClaimType claimType,
+      Long id,
+      Optional<DateRangeParam> lastUpdated,
+      Optional<DateRangeParam> serviceDate,
+      boolean excludeSamhsa) {
+    this.claimTransformer = requireNonNull(claimTransformer);
+    this.claimType = requireNonNull(claimType);
+    this.id = requireNonNull(id);
+    this.lastUpdated = lastUpdated;
+    this.serviceDate = serviceDate;
+    this.excludeSamhsa = excludeSamhsa;
+  }
+
+  /**
+   * Sets the {@link #entityManager}; defined as scope=prototype to get unique JPA {@link
+   * EntityManager} per thread.
+   *
+   * @param entityManager a JPA {@link EntityManager} connected to the application's database
+   */
+  @PersistenceContext
+  @Scope("prototype")
+  public void setEntityManager(EntityManager entityManager) {
+    this.entityManager = entityManager;
+  }
+
+  /**
+   * Sets the {@link #includeTaxNumbers} which will turn on processing of NPI tax number info.
+   *
+   * @param includeTaxNumbers {@link boolean} to enable/disable NPI tax number processing.
+   */
+  public void setIncludeTaxNumbers(boolean includeTaxNumbers) {
+    this.includeTaxNumbers = includeTaxNumbers;
+  }
+
+  /**
+   * ExecutorService will invoke the task.
+   *
+   * @return the results for the task.
+   */
+  @Override
+  public PatientClaimsEobTaskTransformer call() {
+    LOGGER.debug("TransformPatientClaimsToEobTaskpwd.call() started for {}", id);
+    try {
+      eobs.addAll(transformToEobs(findClaimTypeByPatient()));
+      if (excludeSamhsa) {
+        filterSamhsa(eobs);
+      }
+    } catch (NoResultException e) {
+      LOGGER.error((taskException = e).getMessage(), e);
+    } catch (Exception e) {
+      // keep track of the Exception so we can provide to caller.
+      LOGGER.error((taskException = e).getMessage(), e);
+    }
+    return this;
+  }
+
+  /**
+   * Transform a list of claims to a list of {@link ExplanationOfBenefit} objects.
+   *
+   * @param claims the claims/events to transform
+   * @return the {@link ExplanationOfBenefit} instances, one per claim/event
+   */
+  @Trace
+  private List<ExplanationOfBenefit> transformToEobs(List<?> claims) {
+    return claims.stream().map(c -> transformEobClaim(c)).collect(Collectors.toList());
```

Comment:
Not a big deal but this can be shortened now by calling `toList()` immediately after `map()` instead of using `collect()`.